src/main/java/edu/harvard/iq/dataverse/engine/command/impl/PublishDatasetCommand.java

```diff
@@ -196,6 +199,24 @@ private void verifyCommandArguments() throws IllegalCommandException {
                 throw new IllegalCommandException("Cannot release as minor version. Re-try as major release.", this);
             }
         }
-    }   
+    }
+    
+    
+    @Override
+    public boolean onSuccess(CommandContext ctxt, Object r) {
+        Dataset dataset = null;
+        try{
+            dataset = (Dataset) r;
+        } catch (ClassCastException e){
+            dataset  = ((PublishDatasetResult) r).getDataset();
+        }
+
+        if (dataset != null) {
```

Comment:
I guess this is just to be extra safe, but would this ever be false? (since we're already in OnSuccess, that means the execute succeeded, no?

Reply:
I guess in case something else still goes wrong, in some other way? - idk.